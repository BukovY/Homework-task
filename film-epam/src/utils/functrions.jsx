export const genresIndexToString = (arr, map) => {
  const genresToRender = [];
  for (let i of map) {
    if (arr.indexOf(i.id) !== -1) {
      genresToRender.push(i.name);
    }
  }
  return genresToRender;
};

export const minToTime = (num) => {
  let min = num % 60;
  let hour = (num - min) / 60;
  return String(hour) + ":" + String(min);
};

export const getPhotoCard = (path) => {
  return path === null || path === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w1280/${path}`;
};

export const getPeopleCard = (img) => {
  return img === null || img === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getFilmCover = (img) => {
  return img === null || img === undefined
    ? "https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getPaginationRange = (maxPage) =>
  Array(maxPage)
    .fill(1)
    .reduce((prev, next, index) => {
      return [...prev, index + 1];
    }, []);

export const getIndexLanguage = (lang) => {
  return lang === "EN" ? 0 : lang === "RU" ? 1 : 2;
};

export const matchOnlyNumber = (str) => /^\d+$/.test(str);
