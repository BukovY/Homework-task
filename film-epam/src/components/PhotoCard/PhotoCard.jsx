import React from "react";
import styles from "./PhotoCard.module.sass";
import { getPhotoCard } from "../../utils/functrions";

const PhotoCard = ({ path }) => {
  return (
    <img src={getPhotoCard(path)} className={styles.photo_card} alt="PhotoCard" />
  );
};
export default PhotoCard;
