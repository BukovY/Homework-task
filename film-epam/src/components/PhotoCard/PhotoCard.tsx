import React, { FC } from "react";
import { getPhotoCard } from "../../utils/functrions";
import { makeStyles } from "@material-ui/core/styles";
type PhotoCardProps = {
  path: string;
};

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    height: "auto",
    borderRadius: "10px",
  },
}));

export const PhotoCard: FC<PhotoCardProps> = ({ path }) => {
  const classes = useStyles();
  return (
    <img
      src={getPhotoCard(path)}
      className={classes.root}
      alt="PhotoCard"
    />
  );
};
