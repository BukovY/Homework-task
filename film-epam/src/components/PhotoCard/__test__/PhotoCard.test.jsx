import React from "react";
import rerender from "react-test-renderer";
import { PhotoCard } from "../PhotoCard";
import { wrapperProvider } from "../../../utils/functrions";

describe("Testing <People/> Component", () => {
  it("renders correctly", () => {
    const tree = rerender
      .create(wrapperProvider(<PhotoCard path={null} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
