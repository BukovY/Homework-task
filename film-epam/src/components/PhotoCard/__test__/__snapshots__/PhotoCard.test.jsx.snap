// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <People/> Component renders correctly 1`] = `
<img
  alt="PhotoCard"
  className="makeStyles-root-1"
  src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
/>
`;
